### ISSUE TITLE
What is predict_proba() for?

### ISSUE THREAD

**Issue by peastman** [deepchem/deepchem#1214]
For each of the `predict*()` methods in TensorGraph, we have a second completely identical method where `predict` is changed to `predict_proba` in the name.  What are these for?  Are they just there for historical reasons?  Should we remove them?

**Comment by rbharath** [comment]
I believe that `predict_proba` was used by the classifier models at some point. This no longer appears to be the case (https://github.com/deepchem/deepchem/blob/master/deepchem/models/tensorgraph/fcnet.py#L171 calls the superclass method). I think we should be fine getting rid of it.

**Comment by miaecle** [comment]
Is it also related to the compatibility of sklearn models? For random forest and logistic regression these two methods should be different.

**Comment by peastman** [comment]
Actually, MultitaskClassifier does implement them differently.  `predict_proba()` just calls the superclass method, but `predict()` overrides it to return the argmax.  This is the sort of inconsistency I've been fighting against, where the outputs returned by `predict()` don't correspond to the actual outputs of the model.  Just to make things worse, it only overrides `predict()`, not `predict_on_batch()` or `predict_on_generator()`.  So those methods return different kinds of values.

**Comment by peastman** [comment]
And to make it even worse: until I changed it last month in #1106, the output of MultitaskClassifier was logits, not probabilities.  So `predict_proba()` didn't actually return probabilities!

**Comment by peastman** [comment]
I suggest we clean all of this up, standardizing on the following principles.

1. Models should almost never override any of the predict methods.  Whatever the outputs of the graph are, that's what `predict()` should return.  Some very atypical models may add their own prediction methods (e.g. `SeqToSeq`), but even then they should be new methods, not overriding the existing ones.

2. For classification models, the outputs should always be class probabilities.  Not logits and not class indices.  The former aren't generally useful to users, and the latter are trivial to compute with a call to argmax if that's what you want.

3. We get rid of the `predict_proba()` methods.  For regression models they're meaningless, and for classification models `predict()` is already returning probabilities.

**Comment by rbharath** [comment]
I'm on-board with all these suggested predictions. Will make behavior more intuitive for users.
